src/engine/guidance/collapse_turns.cpp

```diff
@@ -477,6 +478,64 @@ RouteSteps collapseTurnInstructions(RouteSteps steps)
     return steps;
 }
 
+std::vector<RouteStep> suppressSegregated(std::vector<RouteStep> steps)
```

Comment:
this entire function cannot be in here, if we want to merge this and continue working. The segregated flag should supplement turn compression, not do additional work on non-compressed turns.